### PR TITLE
Fixes for telemetry

### PR DESCRIPTION
**Pull request by superlime**
- Properly call WindowsAppInitializer to set up telemetry providers
- Add event for application launch
- Add standardized events for connections
- Add custom TelemetryInitializer to fix device.type and add device.family

### CODE REVIEW

File: Shield/MainPage.xaml.cs [hunk 1]
```diff
@@ -436,28 +446,38 @@ private void InitializeManager()
...
             {
                 bool worked = false;
+                connectionStopwatch.Reset();
+                connectionStopwatch.Start();
+
                 await dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
                 {
                     appSettings.CurrentConnectionState = (int)ConnectionState.Connecting;
+                    App.Telemetry.Context.Properties["connection.state"] = ConnectionState.Connecting.ToString();
+                    App.Telemetry.Context.Properties["connection.name"] = selectedConnection.DisplayName;
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml.cs]: potentially contains PII - this line needs to disappear quickly

File: Shield/MainPage.xaml.cs [hunk 2]
```diff
@@ -436,28 +446,38 @@ private void InitializeManager()
...
+                    App.Telemetry.Context.Properties["connection.detail"] = GetConnectionDetail(selectedConnection);
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml.cs]: this looks to have too much info also (remote name, other serialized parms)

File: Shield/MainPageHandleMessage.cs [hunk 3]
```diff
@@ -885,13 +885,13 @@ private void ToggleSensors(SensorMessage sensorsMessage)
                         throw new UnsupportedSensorException("Compass does not exist");
                     }
 
-                    App.Telemetry.TrackEvent("Sensor", new Dictionary<string, string> { { "M", sensorItem.M.Value.ToString() } });
-                    Sensors.SensorSwitches.M = sensorItem.M.Value;
+                    App.Telemetry.TrackEvent("Sensor", new Dictionary<string, string> { { "sensor.type", sensorsMessage.Type.ToString() }, { "sensor.value", sensorItem.M.Value.ToString() } }); Sensors.SensorSwitches.M = sensorItem.M.Value;
```
Comment (JimGaleForce) [inline Shield/MainPageHandleMessage.cs]: missing return here

File: Shield/MainPage.xaml.cs [hunk 4]
```diff
@@ -607,6 +626,37 @@ public void UpdateDestinations()
...
+                var source = connection.Source as EndpointPair;
+                return string.Format("{0}:{1}", source.RemoteHostName, source.RemoteServiceName);
+            }
+            else if (connection.Source is RemotePeer)
+            {
+                var source = connection.Source as RemotePeer;
+                return source.ToString();
+            }
+            else if (connection.Source is string)
+            {
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml.cs]: these can be combined ... the .ToString() operates even if it is already a string

File: Shield/App.xaml.cs [hunk 5]
```diff
@@ -55,6 +56,13 @@ sealed partial class App : Application
         /// </summary>
         public App()
         {
+            WindowsAppInitializer.InitializeAsync()
```
Comment (JimGaleForce) [inline Shield/App.xaml.cs]: probably should be try...catched so that a bad telemetry component/connection won't take down the app

Comment (superlime) [inline Shield/App.xaml.cs]: It looks like exceptions inside AppInsights are automatically handled.  None of the best practices documentation suggests doing this, and as a test I just added 'throw new Exception("foo");' inside UwpDeviceTelemetryInitializer, and while the exceptions were thrown ("Exception thrown: 'System.Exception' in Shield.exe" showed up in debug output), it did not have any effect on the app from the user's perspective (still running, UI fully functional, etc).

I think this one's fine to leave as is, unless you've got a specific scenario in mind?

File: Shield/MainPageHandleMessage.cs (hunk 3 above)
Comment (superlime) [inline Shield/MainPageHandleMessage.cs]: Not sure how that got removed, good catch!

File: Shield/MainPage.xaml.cs (hunk 1 above)
Comment (superlime) [inline Shield/MainPage.xaml.cs]: Will change to return just a hash, good catch.

File: Shield/MainPage.xaml.cs (hunk 2 above)
Comment (superlime) [inline Shield/MainPage.xaml.cs]: Will change to return just a hash, good catch.

**Pull request merged**